include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -223,108 +218,155 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 					std::array<uint64_t, 4> hash = {};
```

Comment:
if there are fields that need to be properly initialized, then only the constructor should be public